nexus/src/app/instance.rs

```diff
@@ -1052,6 +1091,264 @@ impl super::Nexus {
+                            }
+                            ipnetwork::IpNetwork::V6(network) => {
+                                dpd_client
+                                    .nat_ipv6_delete(
+                                        &network.ip(),
+                                        *target_ip.first_port,
+                                    )
+                                    .await
+                            }
+                        }
+                        .map_err(|e| {
+                            Error::internal_error(&format!(
+                                "failed to clear dpd entry: {e}"
+                            ))
+                        })?;
+                    } else {
+                        info!(log,
+                          "nat entry with expected internal ip exists, continuing";
+                          "instance_id" => %instance_id,
+                          "external_ip" => ?target_ip,
+                          "existing_entry" => ?existing);
+
+                        continue;
+                    }
+                }
+                Err(e) => {
+                    if e.status() == Some(http::StatusCode::NOT_FOUND) {
+                        info!(log, "no nat entry found for: {target_ip:#?}");
+                    } else {
+                        return Err(Error::internal_error(&format!(
+                            "failed to query dpd: {e}"
+                        )));
+                    }
+                }
+            }
+
+            info!(log, "creating nat entry for: {target_ip:#?}");
+            let nat_target = dpd_client::types::NatTarget {
+                inner_mac: dpd_client::types::MacAddr {
+                    a: mac_address.into_array(),
+                },
+                internal_ip: *sled_ip_address.ip(),
+                vni: vni.into(),
+            };
+
+            match target_ip.ip {
+                ipnetwork::IpNetwork::V4(network) => {
+                    dpd_client
+                        .nat_ipv4_create(
+                            &network.ip(),
+                            *target_ip.first_port,
+                            *target_ip.last_port,
+                            &nat_target,
+                        )
+                        .await
+                }
+                ipnetwork::IpNetwork::V6(network) => {
+                    dpd_client
+                        .nat_ipv6_create(
+                            &network.ip(),
+                            *target_ip.first_port,
+                            *target_ip.last_port,
+                            &nat_target,
+                        )
+                        .await
+                }
+            }
+            .map_err(|e| {
+                Error::internal_error(&format!(
+                    "failed to create nat entry: {e}"
+                ))
+            })?;
+
+            debug!(log, "creation of nat entry successful for: {target_ip:#?}");
```

Comment:
why is this debug when all the others in this function are info? Personally I'd be interested in the success message not being filtered out depending on log level :)

Reply:
Fixed in df655b34.